File: Java-Core/src/main/java/presto/java/JavaExpressionList.java
package presto.java;

import presto.utils.CodeWriter;
import presto.utils.ObjectList;

public class JavaExpressionList extends ObjectList<JavaExpression> {

	private static final long serialVersionUID = 1L;

	public JavaExpressionList() {
	}
	
	public JavaExpressionList(JavaExpression expression) {
		this.add(expression);
	}

	public void toDialect(CodeWriter writer) {
		if(this.size()>0) {
			for(JavaExpression exp : this) {
				exp.toDialect(writer);
				writer.append(", ");
			}
			writer.trimLast(2);
		}
	}
	
}

File: Java-CodeServer/src/main/resources/js/prompto/java/JavaExpressionList.js
var ObjectList = require("../utils/ObjectList").ObjectList;

function JavaExpressionList(expression) {
	ObjectList.call(this);
	expression = expression || null;
	if(expression!==null) {
		this.add(expression);
	}
	return this;
}

JavaExpressionList.prototype = Object.create(ObjectList.prototype);
JavaExpressionList.prototype.constructor = JavaExpressionList;

JavaExpressionList.prototype.toDialect = function(writer) {
    if(this.length>0) {
        for(var i=0;i<this.length;i++) {
            var exp = this[i];
            exp.toDialect(writer);
            writer.append(", ");
        }
        writer.trimLast(2);
    }
};

exports.JavaExpressionList = JavaExpressionList;